pkg/backup/remap_crd_version_action_test.go

```diff
@@ -31,27 +31,26 @@ import (
 	velerotest "github.com/vmware-tanzu/velero/pkg/test"
 )
 
+//TODO: update tests
+// Need a fake client for the v1beta1 CRDs
+// TODO: How do we get the elasticsearch & crossplane CRDs in here to run w/o a cluster?
 func TestRemapCRDVersionAction(t *testing.T) {
 	backup := &v1.Backup{}
-	a := NewRemapCRDVersionAction(velerotest.NewLogger())
+	v1Client := new(apiextv1fakes.FakeApiextensionsV1)
+	betaClient := new(apiextv1beta1fakes.FakeApiextensionsV1beta1)
+
+	a := NewRemapCRDVersionAction(velerotest.NewLogger(), betaClient.CustomResourceDefinitions())
 
 	t.Run("Test a v1 CRD without any Schema information", func(t *testing.T) {
 		b := builder.ForV1CustomResourceDefinition("test.velero.io")
 		// Set a version that does not include and schema information.
 		b.Version(builder.ForV1CustomResourceDefinitionVersion("v1").Served(true).Storage(true).Result())
 		c := b.Result()
-		obj, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&c)
-		require.NoError(t, err)
 
-		item, _, err := a.Execute(&unstructured.Unstructured{Object: obj}, backup)
+		// Add the v1 resource to the API server so we can fetch it as v1beta1
+		_, err := v1Client.CustomResourceDefinitions().Create(c)
```

Comment:
This current causes a nil pointer dereference panic, probably because the v1Client isn't being instantiated correctly. However, the fake package for it doesn't include a `NewSimpleClientSet` like other generated clients.